src/clj/harja/kyselyt/indeksit.clj

```diff
@@ -1,7 +1,72 @@
+          urakan-alkuvuosi (-> alkupvm pvm/joda-timeksi pvm/suomen-aikavyohykkeeseen pvm/vuosi)
+          urakan-loppuvuosi (-> loppupvm pvm/joda-timeksi pvm/suomen-aikavyohykkeeseen pvm/vuosi)
+          vertailu-kk-mhu (fn [urakan-akuvuosi]
+                            (cond
+                              ;; HOX!!
+                              ;; Jos tähän tulee sopimusmuutoksia, tee samat muutokset sql funktioon: indeksikorjaa
+                              ;;
+                              ;; 2023 ja jälkeen alkavilla urakoilla käytetään indeksin tarkastelukuukautena elokuuta
+                              (>= urakan-akuvuosi 2023) 8
+                              ;; Muihin aikoihin alkavilla urakoilla käytetään tarkastelukuukautena syyskuuta
+                              :else 9))
+          perusluku (:perusluku (first (hae-urakan-indeksin-perusluku db {:urakka-id urakka-id})))
+          indeksiluvut-urakan-aikana (when perusluku
+                                       (sequence
+                                         (comp (filter (fn [{:keys [kuukausi vuosi]}]
+                                                         (and (= (vertailu-kk-mhu urakan-alkuvuosi) kuukausi)
+                                                           (>= vuosi urakan-alkuvuosi))))
+                                           (remove (fn [{:keys [vuosi]}]
+                                                     (>= vuosi urakan-loppuvuosi)))
+                                           (map (fn [{:keys [arvo vuosi]}]
+                                                  {:vuosi vuosi
+                                                   ;; Halutaan numero kolmen desimaalin tarkkuudella. Pelataan sen varaan, että indeksikerroin
+                                                   ;; Ei nouse yli kymmenen, jolloin with-precision 4 riittää.
+                                                   ;; Ratkaisu pyöristää indeksikerrointa. Tämä on sovittu käytäntö ELYissä ja perustuu myös siihen,
+                                                   ;; että tilastokeskus ilmaisee indeksikertoimen kolmella desimaalilla (prosentin kymmenyksen tarkkuudella).
+                                                   :indeksikerroin (pyorista (with-precision 4 (/ arvo perusluku)) 3)})))
+                                         (hae-indeksi db {:nimi indeksi})))
+          urakan-indeksien-maara (count indeksiluvut-urakan-aikana)]
+      (if (= 5 urakan-indeksien-maara)
+        (vec indeksiluvut-urakan-aikana)
+        (mapv (fn [index]
+                (if (empty? indeksiluvut-urakan-aikana)
+                  ;;Palautetaan nil indeksikertoimeksi urakoille, jotka eivät ole vielä alkaneet.
+                  nil
+                  ;; Palautetaan indeksit vain hoitovuosille, joilla on indeksejä.
+                  ;; Lopuille hoitovuosille nil.
+                  (nth indeksiluvut-urakan-aikana index nil)))
+          (range 0 5))))))
```

Comment:
Onko tämä koodia, joka toimii vain 5 vuotisille MHU:ille?

Reply:
Erittäin hyvä pointti! 👍  Tuota funktiota käytetään tuon vanhan kustiksen kautta useassa paikassa. Tässä siis vain copypastesin sen uuteen paikkaan ja hieman muutin funktion parametreja. Teen tähän korjauksen samalla sekä tähän, minkä copypastesin, että siihen alkuperäiseen.